src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -1029,11 +1031,13 @@ void NetworkOPsImp::apply (std::unique_lock<std::mutex>& batchLock)
     batchLock.unlock();
 
     {
-        auto lock = make_lock(app_.getMasterMutex());
+        auto lock = make_lock(app_.getMasterMutex(), std::defer_lock);
         bool changed = false;
         {
+            auto& mut = m_ledgerMaster.peekMutex();
+            std::lock(lock, mut);
             std::lock_guard <std::recursive_mutex> lock (
-                m_ledgerMaster.peekMutex());
+                mut, std::adopt_lock);
```

Comment:
This `lock` shadows the variable at 1034, which is just confusing. Moreover, I just find this whole sequence of code to be unreadable.

At 1034 we defer locking the master mutex. At 1038 we lock the master mutex and the the ledger master mutex (in a well-defined order) and in 1040 we "adopt" the ledger master mutex into a lock guard?

There has to be a cleaner way of doing all this stuff.

Reply:
I hadn't noticed the shadowing.  Good catch.  I'll rename one of them.

An alternative formulation would be:
```
        auto lock = make_lock(app_.getMasterMutex(), std::defer_lock);
        bool changed = false;
        {
            auto lock2 = make_lock(m_ledgerMaster.peekMutex(), std::defer_lock);
            std::lock(lock, lock2);
```
Neither has a significant performance or behavioral advantage over the other, so it is up to which ever we find more readable.  I invented this stuff and so am probably too close to it for subjective positions on readability (I know it _too_ well).

The whole adopt/defer dance is to put RAII wrappers on the mutexes to make them exception safe, and yet have `std::lock` actually do the locking of them to combat deadlock.  And another minor complication in this example is that we want to limit the scope of the inner lock more than that of the outer lock.

Reply:
I do find the alternative formulation a bit easier to think about.  Maybe we could also give the two locks slightly more meaningful names than `lock` and `lock2`?  Possibly something like this...
```
    {
        auto lockMaster = make_lock(app_.getMasterMutex(), std::defer_lock);
        bool changed = false;
        {
            auto lockLedger =
                make_lock(m_ledgerMaster.peekMutex(), std::defer_lock);
            std::lock(lockMaster, lockLedger);

            app_.openLedger().modify(
                [&](OpenView& view, beast::Journal j) ...
```

Reply:
I've made these changes and force-pushed them in so that we don't later have to figure out which commit needs folding.